### ISSUE TITLE
`stripe events resend` doesn't work with -v (aka --stripe-version)

### ISSUE THREAD

**Issue by geodn** [stripe/stripe-cli#1091]
_The more information we have the easier it is for us to help. Feel free to remove any sections that might not apply_

### Issue
I tried to resend an event with a different stripe version (-v 2022-11-15) and it doesn't use the version I specify. It still uses the old version 2020-08-27.

### Expected Behavior
Event is re-sent with the new stripe version.

### Steps to reproduce
Take an event sent with version 2020-08-27

Run
`stripe events resend [eventid] --stripe-version 2022-11-15 --webhook-endpoint=[endpoint]`

(I also passed in --api-key here)

Look at the api_version field in the request.

### Traceback
_Share any debug output that was given by the CLI_

### Environment
macOS

**Comment by corydambach** [comment]
Yeah, this doesn't work for me either.  Not sure if it is working as intended or not.  Was able to work around it.

**Comment by toddams** [comment]
it still happens, @corydambach how were u able to workaround?

**Comment by giaphoang** [comment]
I believe this issue is already resolved in the CLI build tested here.

Exact CLI version and source revision:

```text
$ ./bin/stripe version
stripe version master

$ git describe --tags --always --dirty
v1.50.11-8-g4459f9e

$ git rev-parse --short HEAD
4459f9e
```

Command:

```bash
./bin/stripe events resend evt_123 \
  --stripe-version 2022-11-15 \
  --webhook-endpoint we_123 \
  --dry-run
```

Exact output:

```text
▸ Running in Testing Startup · sandbox (acct_1UEwdaFexdnvP6mz)
{
  "dry_run": {
    "method": "POST",
    "url": "https://api.stripe.com/v1/events/evt_123/retry",
    "params": {
      "webhook_endpoint": "we_123"
    },
    "headers": {
      "Authorization": "Bearer oak_live************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************0fMP",
      "Content-Type": "application/x-www-form-urlencoded",
      "Stripe-Context": "acct_1UEwdaFexdnvP6mz",
      "Stripe-Livemode": "false",
      "Stripe-Version": "2022-11-15"
    }
  }
}
```

The important result is:

```text
"Stripe-Version": "2022-11-15"
```

The command exited with status `0`. This shows that the tested `stripe version master` build, source revision `4459f9e`, forwards the requested version header to the resend request. The original report may have involved older CLI/API behavior, or the inspected `api_version` may describe the original event or webhook endpoint rather than the resend request header.

No credentials or live API mutation were used; this was a dry run.

**Comment by corydambach** [comment]
@toddams I must say it was quite a while ago when I looked at it.  
If I recall, an event is fixed at generation time to whatever API version you had set at that time.  
I think I had to make my endpoint accept it as the CLI would let me send it (using the old api ver).  
If I remember more I'll get back to you.
